### PR TITLE
Refactoring isGIF() as a utility function 

### LINKED ISSUE #1501: refactoring isGIF() as a utility function
### Please describe the problem (or idea)
@jywarren's comment:
Refactoring isGIF() as a utility function so the parameters don't have to change for the function [here](https://github.com/publiclab/image-sequencer/pull/1404#discussion_r365530741)

### Please show us where to look

https://beta.sequencer.publiclab.org

### What's your PublicLab.org username?

> This can help us diagnose the issue:

### Browser, version, and operating system

> Many bugs are related to these -- please help us track it down and reproduce what you're seeing!

****

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site.

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by blurry-x-face**
Fixes #1501 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below
* [x] Insert-step functionality is working correct as expected.
> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software
Please make sure to get at least two reviews before asking for merging the PR as that would make the PR more reliable on our part
Thanks!

### CODE REVIEW

**Comment by blurry-x-face** [comment]
@keshav234156 Do you know why UI tests are failing?
@HarshKhandeparkar ?

File: src/util/isGif.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = function(url){
+    
+  const toMatch = url.match('data:image/(.*);');
```
Comment (HarshKhandeparkar) [inline src/util/isGif.js]: It seems this function is glitching. I think URL here is undefined in some cases. Please check that.

Comment (blurry-x-face) [reply src/util/isGif.js]: :thinking: ok, on it!

Comment (blurry-x-face) [reply src/util/isGif.js]: Yes you were right, I fixed it, now tests are running fine, Thanks!!!

**Comment by blurry-x-face** [comment]
@publiclab/is-reviewers please review this

File: src/util/isGif.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = function(url){
+  if(url.match){
+    const toMatch = url.match('data:image/(.*);') || 'png';
```
Comment (HarshKhandeparkar) [inline src/util/isGif.js]: This is a regex right? If that is the case, then can you use regex literals `//`? Put the regex between `/` and `/`.

Comment (blurry-x-face) [reply src/util/isGif.js]: Yes it is working fine, but I'll add a `\` before `/`.

Comment (blurry-x-face) [reply src/util/isGif.js]: I'm sorry could you explain a bit :sweat_smile: I don't understand!

Comment (HarshKhandeparkar) [reply src/util/isGif.js]: JavaScrip has dedicated regex literals just like string literals and template literals
```js
const string = 'This is a string literal. It is just a string'
const templateString = `This is a template literal.
It is a string with superPowers ${variable}`

const regex = /^g$/ig; // This is a regex literal. It is used to store regex (along with options like global (`g`) and case insensitive (`i`).

Comment (blurry-x-face) [reply src/util/isGif.js]: Ohh! I did not knew that! Thanks! :astonished:

Comment (blurry-x-face) [reply src/util/isGif.js]: Done!

**Comment by jywarren** [comment]
Hmm, this is showing up oddly in my interface. Maybe GitHub UI changed? Can you rebase this, it's not offering me an easy button to do so myself!

Thank you! this is great.

**Comment by blurry-x-face** [comment]
@jywarren Done!

**Comment by jywarren** [comment]
Great! Thank you!!!

**Pull request merged**